bundles/io.openliberty.tools.eclipse.ui/src/io/openliberty/tools/eclipse/ui/launch/LaunchConfigurationDelegateLauncher.java

```diff
@@ -72,12 +74,14 @@ public void launch(ILaunchConfiguration configuration, String mode, ILaunch laun
         display.syncExec(new Runnable() {
             public void run() {
                 try {
-                    IProject iProject = Utils.getActiveProject();
-                    if (iProject == null) {
-                        iProject = devModeOps.getSelectedDashboardProject();
-                    }
+                    validateProjectsMatch(configuration);
```

Comment:
The validation looks ok, but this change will only apply to cases where you have the configuration dialog open, in which it would  have been nice to just set the error message on the dialog itself, which will prevent further progress (i.e. have this check be part of the validation we do for the dialog. After all you can get to both the project name in the config and get the active project).
I also don't see any other validation being done for the start and startInContainer actions. I believe this change has to happen in start and in startInContainer action classes.

Reply:
Addressed in later commit